mlflow/projects/__init__.py

```diff
@@ -145,7 +147,29 @@ def _run(uri, experiment_id, entry_point="main", version=None, parameters=None,
             work_dir=work_dir, entry_point=entry_point, parameters=parameters,
             experiment_id=experiment_id,
             use_conda=use_conda, storage_dir=storage_dir, run_id=active_run.info.run_id)
-    supported_backends = ["local", "databricks"]
+    elif backend == "kubernetes":
```

Comment:
What's the behavior if the `synchronous` flag is equal to false. If it's not actually async we should throw an exception here.

Reply:
It's the same behavior as `mode == databricks`, it ignores the flag and runs synchronous. AFAIK this flag is only used in local mode, isn't it?

Reply:
The point about throw an exception is that today the sychronous parameter gets it's value from cli.py in the block below:
```
    try:
        projects.run(
            uri,
            entry_point,
            version,
            experiment_name=experiment_name,
            experiment_id=experiment_id,
            parameters=param_dict,
            backend=backend,
            backend_config=backend_config,
            use_conda=(not no_conda),
            storage_dir=storage_dir,
            synchronous=backend == "local" or backend is None,
            run_id=run_id
        )
    except projects.ExecutionException as e:
        _logger.error("=== %s ===", e)
        sys.exit(1)
```

So this will be always false for `backends` different than `local`. I think that we should let it as it is now. What do you think?

Reply:
I see. I overlooked the fact that it is not used if `mode=databricks`. Makes sense to leave it for now then.

BTW, did you see this comment yet? https://github.com/mlflow/mlflow/pull/1181#discussion_r286227289

Reply:
Hey @andrewmchen ! As we spoke earlier, I changed the implementation to use the `docker_env` again.

Reply:
Just need to update the docs

Reply:
done @andrewmchen ! I think it's ready now :)